### ISSUE TITLE
Add support for time-varying ascertainment

### ISSUE THREAD

**Issue by seabbs** [epiforecasts/EpiNow2#792]
As the title. On a single data source this could/would only be in the form of a user input but if any additional complexity (i.e. multiple data sources) it could be fit.

**Comment by seabbs** [comment]
An example use case could be if you know the number of hospitals that are reporting and this varies with time.

**Comment by kgostic** [comment]
Would you just specify the change point and let the model infer the scaling factor? Or would it be more like you have to input some proxy for the time-varying ascertainment across the entire time period?

**Comment by seabbs** [comment]
So I think you could do either and I think it would make sense to: 

1. Most general case as a feature (i.e its all priros that vary over time)
2. Add a feature to make the fixed input special case
3. Add a helper to do piecewise changes

Starting with 1. makes sense as this then naturally extends to the planned model with multiple time series support

**Comment by sbfnk** [comment]
In principle this could be supported as part of the proposed changes in https://github.com/epiforecasts/EpiNow2/issues/525 and https://github.com/epiforecasts/EpiNow2/issues/600 if ever going down those routes.

**Comment by seabbs** [comment]
Yes agree. I was wondering about a easier version in the first instance though which could be supported right now with a special argument (for case 2.)

**Comment by sbfnk** [comment]
To be addressed following #547 with another column as discussed in #839.

**Comment by seabbs** [comment]
> To be addressed following https://github.com/epiforecasts/EpiNow2/issues/547 with another column as discussed in https://github.com/epiforecasts/EpiNow2/pull/839.

This can be addressed now right? 

If I am remembering correctly we think we want to give special support to this in the first instance with a optional additional column and plan to add more complex functionality to the prior interface to offer wider support at some point in the future (i.e https://github.com/epiforecasts/EpiNow2/issues/600)?
